streampark-console/streampark-console-service/src/test/java/org/apache/streampark/console/core/service/SavePointServiceTest.java

```diff
@@ -0,0 +1,163 @@
+
+    // Test for (StreamPark job Or FlinkSQL job) with application config just disabled checkpoint.
+    ApplicationConfig appCfg = new ApplicationConfig();
+    appCfg.setId(appCfgId);
+    appCfg.setAppId(appId);
+    appCfg.setContent("state.savepoints.dir=hdfs:///test");
+    appCfg.setFormat(ConfigFileType.PROPERTIES.getValue());
+    configService.save(appCfg);
+    assertThat(savePointServiceImpl.getSavepointFromAppCfgIfStreamParkOrSQLJob(app)).isNull();
+
+    // Test for (StreamPark job Or FlinkSQL job) with application config and enabled checkpoint and
+    // configured value.
+
+    // Test for non-value for CHECKPOINTING_INTERVAL
+    assertThat(savePointServiceImpl.getSavepointFromAppCfgIfStreamParkOrSQLJob(app)).isNull();
+
+    // Test for configured CHECKPOINTING_INTERVAL
+    appCfg.setContent(
+        DeflaterUtils.zipString(
+            "state.savepoints.dir=hdfs:///test\n"
+                + String.format("%s=%s", CHECKPOINTING_INTERVAL.key(), "3min")));
+    configService.updateById(appCfg);
+    Effective effective = new Effective();
+    effective.setTargetId(appCfg.getId());
+    effective.setAppId(appId);
+    effective.setTargetType(EffectiveType.CONFIG.getType());
+    effectiveService.save(effective);
+    assertThat(savePointServiceImpl.getSavepointFromAppCfgIfStreamParkOrSQLJob(app))
+        .isEqualTo("hdfs:///test");
+  }
+
+  @Test
+  void testGetSavepointFromDeployLayer() throws JsonProcessingException {
+    SavePointServiceImpl savePointServiceImpl = (SavePointServiceImpl) savePointService;
+    Long appId = 1L;
+    Long idOfFlinkEnv = 1L;
+    Long teamId = 1L;
+    Application application = new Application();
+    application.setId(appId);
+    application.setTeamId(teamId);
+    application.setVersionId(idOfFlinkEnv);
+    application.setExecutionMode(ExecutionMode.YARN_APPLICATION.getMode());
+    applicationService.save(application);
+
+    FlinkEnv flinkEnv = new FlinkEnv();
+    flinkEnv.setFlinkName("mockFlinkName");
+    flinkEnv.setFlinkHome("/tmp");
+    flinkEnv.setId(idOfFlinkEnv);
+    flinkEnv.setVersion("1.15.3");
+    flinkEnv.setScalaVersion("2.12");
+    flinkEnv.setFlinkConf(DeflaterUtils.zipString(SAVEPOINT_DIRECTORY.key() + ": hdfs:///test"));
+    flinkEnvService.save(flinkEnv);
+
+    // Test for non-remote mode
+    assertThat(savePointServiceImpl.getSavepointFromDeployLayer(application))
+        .isEqualTo("hdfs:///test");
+
+    // Start the test lines for remote mode
+    Long clusterId = 1L;
+
+    // Test for it without cluster.
+    application.setExecutionMode(ExecutionMode.REMOTE.getMode());
+    application.setFlinkClusterId(clusterId);
+    assertThatThrownBy(() -> savePointServiceImpl.getSavepointFromDeployLayer(application))
+        .isInstanceOf(NullPointerException.class);
+
+    // Ignored.
+    // Test for it with empty config
+    // Test for it with the configured empty target value
+    // Test for it with the configured non-empty target value
```

Comment:
In my limited read, it would be better use mockito to mock the logic and data branches here.
So, what about replacing the `Ignored` with `TODO` tag ?